### DISCUSSION TITLE
IPSec IKEv2 eap-tls connect from macOS Big Sur client

### DISCUSSION THREAD

**Discussion by sergiyd** [strongswan/strongswan#556] [Q&A] [answered]
Hello,
I have StrongSwan server (Linux strongSwan U5.5.1/K4.9.0-16-amd64) with config
    
```
        # left - local (server) side
        left=%any
        leftid=@xxx.xxxxx.com
        leftauth=pubkey
        leftca="C=XX, O=XX, CN=XX"
        leftcert=xxx.cert.der
        leftsendcert=always
        leftsubnet=0.0.0.0/0
        
        # right - remote (client) side
        right=%any
        rightauth=pubkey
        rightsourceip=10.5.0.0/24
        rightdns=x.x.x.x

conn ikev2-pubkey
        keyexchange=ikev2
        auto=add

conn ikev2-eap-tls
        also="ikev2-pubkey"
        rightauth=eap-tls
        rightsendcert=always
        eap_identity=%identity

```

Then I setup VPN connection on macOS:
* client certificate contains Client Authentication ( 1.3.6.1.5.5.7.3.2 ) purpose (--flag clientAuth)
* client certificate contains subject alt. name = VPN server's DNS
* CA chain of client certificate in System and is trusted
* in VPN connection Remote ID the same as Server Address, Local ID is empty
* in VPN connection Authentication Setting none + certificate

When I'm trying to connect, got next:

> 08[CFG] selected peer config 'ikev2-pubkey'
> 08[IKE] no trusted RSA public key found for 'x.x.x.x'

Where x.x.x.x is local IP address or Local ID if specified.

The server does not have any problem with Windows, Android, MikroTik clients.

So, what is wrong with configuration?

_PS
Guess I had the same with MikroTik server and it was solved by change **Remote ID Type** auto -> ignore. Btw have no idea is it secure or not._

**Comment by tobiasbrunner** [comment, marked as answer]
> client certificate contains subject alt. name = VPN server's DNS

That makes no sense. Why would you configure the server's host name in the client certificate? You probably want to use an email address or hostname for the client instead (does not have to be a real address/hostname), which you then also configure as Local ID there.

> in VPN connection Authentication Setting none + certificate

You are not using EAP-TLS if you do that (which is fine unless you explicitly want to use EAP-TLS). To use EAP-TLS you have to select _Certificate_ as user authentication.
